### PR TITLE
VEN-860, VEN-876, VEN-881 | Improve invoice sending UX

### PR DESCRIPTION
**Pull request by ahie**
## Description :sparkles:

- Disable send invoice button once an invoice has been sent
- Add success toast when invoice sending has been processed
- Remove "send invoice" button once invoice has been further processed
- Display "invoice sent" as button label when offer is still pending user interaction
- UnmarkedWsNoticeView: disable buttons that are waiting for server response
- 

### Closes :no_good_woman:

[VEN-860](https://helsinkisolutionoffice.atlassian.net/browse/VEN-860)
[VEN-876](https://helsinkisolutionoffice.atlassian.net/browse/VEN-876)
[VEN-881](https://helsinkisolutionoffice.atlassian.net/browse/VEN-881)

### Automated tests :gear:️

- `src/features/invoiceCard/__tests__/InvoiceCard.test.tsx`
- ` src/common/deleteButton/__tests__/DeleteButton.test.tsx`
- Updated relevant tests.

### Manual testing :construction_worker_man:

- Navigate to "Nostojärjestysilmoitukset"
- Depending on application status, the send invoice button should be displayed as:
  - "Käsitelty": button should be visible and enabled, with text "Lähetä lasku"
  - "Lähetetty": button should be disabled, with text "Tarjous lähetetty"
  - Any other status: should not display the button at all.
- Depending on application status, the delete invoice button should be displayed as:
  - "Käsitelty": button should be visible and enabled, pressing the button should open a confirmation dialog
  - Any other status: should not display the button at all.
- Sending an invoice should display a success toast.
- Deletion button should not be present for WS notices with application status other than "Käsitelty"/"Odottaa käsittelyä"

### CODE REVIEW

File: src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap [hunk 1]
```diff
@@ -341,7 +341,7 @@ exports[`BerthOfferCard renders normally 1`] = `
           <span
             class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
           >
-            Hyväksy ja lähetä tarjous
+            Lähetä lasku
```
Comment (TatuArvela) [inline src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap:344]: This differs from the design for berth offers: https://share.goabstract.com/6ad48b96-78bb-4830-9652-42e3a073f6f7?collectionLayerId=3fdd8e24-7431-47fc-a2d2-367b1d4d6d4b&mode=design

File: src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx [hunk 2]
```diff
@@ -106,14 +114,15 @@ const UnmarkedWsNoticeView = ({
 
       {order && (
         <InvoiceCard
+          applicationStatus={noticeDetails.status}
           buttonsRight={
-            <Button variant="secondary" theme="coat" onClick={handleDeleteLease}>
+            <Button variant="secondary" theme="coat" onClick={handleDeleteLease} disabled={isDeleteLeaseLoading}>
```
Comment (TatuArvela) [inline src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx:119]: Deleting the application or lease shouldn't be possible once the application has been paid.

This should also have a confirmation modal: https://helsinkisolutionoffice.atlassian.net/browse/VEN-881

**Review by TatuArvela — CHANGES_REQUESTED** [review]
Nice improvements! I left a couple of comments to discuss

File: src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx (hunk 2 above)
Comment (ahie) [reply src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx:119]: I'll add the VEN-881 modal to this PR as well. What about the "pending" state? Should the possibility to delete the application/lease be disable for that case as well?

File: src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap (hunk 1 above)
Comment (ahie) [reply src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap:344]: Yeah, unified both berth and winter storage to use "Lähetä lasku". Though "Hyväksy ja lähetä tarjous" is more descriptive. Should I change both to that instead?

File: src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx (hunk 2 above)
Comment (ahie) [reply src/features/unmarkedWsNoticeView/UnmarkedWsNoticeView.tsx]: Done.

File: src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap (hunk 1 above)
Comment (ahie) [reply src/features/applicationView/berthOfferCard/__tests__/__snapshots__/BerthOfferCard.test.tsx.snap]: Implemented as discussed on slack.

File: src/common/applicationDetails/__tests__/ApplicationDetails.test.tsx [hunk 3]
```diff
@@ -130,14 +131,19 @@ describe('ApplicationDetails', () => {
   });
 
   it('renders delete lease button if handleDeleteLease prop is provided', () => {
+    // Using a component with react-modal. Silence error output.
+    jest.spyOn(console, 'error').mockImplementation(() => {
+      /* NO-OP */
+    });
+
```
Comment (TatuArvela) [inline src/common/applicationDetails/__tests__/ApplicationDetails.test.tsx:138]: Would `ReactModal.setAppElement('body');` work here?

File: src/common/deleteButton/__tests__/DeleteButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import ConfirmationModal from '../../confirmationModal/ConfirmationModal';
+import Button from '../../button/Button';
+
+describe('DeleteButton', () => {
+  // Silence react-modal errors
+  beforeEach(() => {
+    jest.spyOn(console, 'error').mockImplementation(() => {
+      /* NO-OP */
+    });
+  });
```
Comment (TatuArvela) [inline src/common/deleteButton/__tests__/DeleteButton.test.tsx:14]: Would `ReactModal.setAppElement('body');` work here?

File: src/common/deleteButton/testUtils.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { ReactWrapper } from 'enzyme';
+
+import DeleteButton from './DeleteButton';
+import Button from '../button/Button';
+import ConfirmationModal from '../confirmationModal/ConfirmationModal';
+
+export const handleDelete = (wrapperContainingDeleteButton: ReactWrapper) => {
+  wrapperContainingDeleteButton.find(DeleteButton).find(Button).simulate('click');
+  wrapperContainingDeleteButton.find(ConfirmationModal).find(Button).last().simulate('click');
+};
```
Comment (TatuArvela) [inline src/common/deleteButton/testUtils.ts:10]: Should we add `"!**/testUtils.ts"` to `package.json jest.collectCoverageFrom` and `"src/**/testUtils.ts"` to `tsconfig.json exclude`?

File: src/features/invoiceCard/InvoiceCard.tsx [hunk 6]
```diff
@@ -57,6 +60,23 @@ const InvoiceCard = ({
         )
     );
 
+  const getSendButton = () => {
```
Comment (TatuArvela) [inline src/features/invoiceCard/InvoiceCard.tsx:63]: Could be called `renderSendButton`, but I'll leave it up to you

File: src/locales/fi.json [hunk 7]
```diff
@@ -91,6 +91,8 @@
       "ENGLISH": "Englanti"
     },
     "terminology": {
+      "harbor": "Satama",
+      "selection": "Valinta",
```
Comment (TatuArvela) [inline src/locales/fi.json:95]: Excellent! Not sure how/why I missed this

**Review by TatuArvela — APPROVED** [review]
Added a few more comments to consider, but looks good to me overall! 👍

File: src/common/applicationDetails/__tests__/ApplicationDetails.test.tsx (hunk 3 above)
Comment (ahie) [reply src/common/applicationDetails/__tests__/ApplicationDetails.test.tsx:138]: Great, this works! Done.

File: src/common/deleteButton/__tests__/DeleteButton.test.tsx (hunk 4 above)
Comment (ahie) [reply src/common/deleteButton/__tests__/DeleteButton.test.tsx:14]: Same here. Done.

File: src/common/deleteButton/testUtils.ts (hunk 5 above)
Comment (ahie) [reply src/common/deleteButton/testUtils.ts:10]: Decided to remove this as it's not really discoverable. Not worth the configuration changes for a one-off function IMO.

File: src/features/invoiceCard/InvoiceCard.tsx (hunk 6 above)
Comment (ahie) [reply src/features/invoiceCard/InvoiceCard.tsx:63]: Changed.

**Pull request merged**